core/trino-main/src/main/java/io/trino/metadata/ViewDefinition.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.metadata;
+
+import io.trino.spi.TrinoException;
+import io.trino.spi.connector.ConnectorViewDefinition;
+import io.trino.spi.security.Identity;
+
+import java.util.List;
+import java.util.Optional;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.trino.spi.StandardErrorCode.INVALID_VIEW;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class ViewDefinition
+{
+    private final String originalSql;
+    private final Optional<String> catalog;
+    private final Optional<String> schema;
+    private final List<ViewColumn> columns;
+    private final Optional<String> comment;
+    private final Optional<Identity> runAsIdentity;
```

Comment:
This is not appropriate for materialized views. "run as" doesn't even exist as a concept for them, just like it doesn't exist for tables.